Mk 32 Pistol, Mark 1 Rifle, Wakizashi, and more

TadJohnson00 opened the pull request:
<!-- The text between the arrows are comments - they will not be visible on your PR. -->
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR <!-- Describe the Pull Request here. What does it change? What other things could this impact? -->

This PR adds some weapons promised as part of the gun rework; it's just a small one since it's my first PR and I'm adding these early so that they can be mapped in sooner.

The Mark 32 "Universal" Pistol is a semi-automatic pistol chambered in .25 caseless, feeding from a twelve-round magazine. It's sold by Nanotrasen as a premium sidearm with superior offensive capabilities, lending itself to the role of a concealed weapon for officials, diplomats, and command staff. Must be mapped in manually.

The Mark 1 Rifle (a spiritual successor to the Garand) is a semi-automatic rifle chambered in .30 light, feeding from an internal five-round magazine that can be loaded from stripper clips or by hand. This reliable and much-loved design also comes in tasteful ceremonial colours for command staff. Both options must be mapped in manually.

The Wakizashi is a short Japanese blade, often worn with a Katana as part of Daisho, literally meaning "big-little". Along with the Wakizashi is a newly-sprited Katana to fit a modern art style, and a ceremonial set of sheaths for the pair of swords. This will be found in the Head of Security's locker.

The Chief Medical Officer is getting a chest rig in tasteful medical-themed colours; now the CMO can carry just about anything, and is not limited to purely medical items in their belt. The CMO chest rig will be found in their locker.

Additionally, the double-barrel shotgun will no longer need to be cycled, allowing for faster takedowns of rowdy bar patrons.

Finally, Syndicate agents will now be able to purchase a box of .30 rifle stripper clips from their uplinks, instead of useless AKM magazines that need to be unloaded manually such that their rounds may be used.

**Media**
<!-- 
If applicable, add screenshots or videos to showcase your PR. Small fixes/refactors are exempt, but all PRs which make ingame changes 
(adding clothing, items, new features, etc) must include ingame media or the PR will not be merged, in accordance with our PR guidelines.
This makes it much easier for us to merge PRs and find media for progress reports. If you include media in your pull request, we 
may potentially use it in the SS14 progress reports, with clear credit given.

Use screenshot software like Window's built in snipping tool, ShareX, Lightshot, or recording software like ShareX (gif), ScreenToGif, or Open Broadcaster Software (cross platform).
If you're unsure whether your PR will require media, ask a maintainer.

Check one of the boxes below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->

![mk32](https://user-images.githubusercontent.com/120885811/230802324-807e9ffc-f1b1-4908-9d13-11d1088dae4f.PNG)
![mark1](https://user-images.githubusercontent.com/120885811/230802329-5430bbf9-7807-4da4-8f59-c1042a2c4d86.PNG)
![katana](https://user-images.githubusercontent.com/120885811/230802332-8a01aa6d-6c31-4874-8de3-ad8ec179ae78.PNG)
![cmo](https://user-images.githubusercontent.com/120885811/230802336-79eea93e-252f-44a9-8014-4c3cd8652007.PNG)

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Here you can fill out a changelog that will automatically be added to the game when your PR is merged
There are 4 icons for changelog entries: add, remove, tweak, fix. I trust you can figure out the rest.

You can put your name after the :cl: symbol to change the name that shows in the changelog (otherwise it takes your GitHub username)
Like so: :cl: PJB

Generally, only put things in changelogs that players actually care about. Stuff like "Refactored X system, no changes should be visible" shouldn't be on a changelog.

For writing actual entries, don't consider the entry type suffix (e.g. add) to be "part" of the sentence:
bad: - add: a new tool for engineers
good: - add: added a new tool for engineers
-->

:cl: TJohnson (Diamora Jensens)
- add: Added Mk 32 Universal, Mark 1 Rifle and Ceremonial variant, resprited katana, added CMO chest rig.
- remove: Removed the need to cycle double-barrel shotguns.

TadJohnson00 committed:
Add Mk 32 Universal, Mk 1 Rifle, Katana Sheath

* Adds Mk 32 Universal - a robust .25 caseless pistol with enhanced offensive capabilities.

* Adds Mark 1 Rifle - a reliable and much-loved design for .30 light rifle. Also comes in tasteful ceremonial colours.

* Adds wakizashi - essentially a smaller katana.

* Adds katana belt - holds both a katana and a wakizashi.

* Adds CMO webbing - a chest rig in medical colours.

* Resprites katana to fit a more modern art style.

* Revises double barrel shotguns - they now automatically cycle.

TadJohnson00 committed:
Added Mk32 Sprites

* Adds the sprites. I forgor

TadJohnson00 committed:
Updated tags

* Added tag for MagazineUniversalCaselessRifle. Maybe this fixes it...

TadJohnson00 committed:
Added belts to head lockers, changed uplink

* Added CMO rig to CMO locker

* Added katana sheath filled to HoS locker

* Changed the .30 rifle mag box to a .30 rifle speed loader box in uplink

TadJohnson00 committed:
Reverted .25 caseless rubber magazine changes

* Reverted a change I made when tired. Will actually fix .25 caseless magazines and references properly later.

TadJohnson00 committed:
Removed erroneous katana sheath sprites

* Removed erroneous sprites messing up the RSI validation.

TadJohnson00 committed:
Merge branch 'master' into master

TadJohnson00 committed:
Merge branch 'Nyanotrasen:master' into master

TadJohnson00 committed:
Removed Filled Katana Sheath from HoS locker

- Removed Filled Katana Sheath from HoS locker. Must now be mapped in manually.

TadJohnson00 committed:
Merge branch 'Nyanotrasen:master' into master

TadJohnson00 committed:
Merge branch 'master' into master

TadJohnson00 committed:
Fix typo in snipers.yml

Changed "anti-material" to "anti-materiel", as it should be

TadJohnson00 committed:
Moved planned additions from upstream files

- Moved code occupying upstream files to Nyanotrasen folders
- Left code that directly overrides upstream files

TadJohnson00 committed:
Merge branch 'master' of https://github.com/TadJohnson00/Nyanotrasen

TadJohnson00 committed:
Fixed file swap issue

Files for caseless mags were swapped, now it should be rectified.

TadJohnson00 committed:
Fixed another typo

Damn these things

TadJohnson00 committed:
RSI file moved

Everything actually works now

Elijahrane commented on Resources/Prototypes/Catalog/Fills/Lockers/heads.yml:
```diff
@@ -120,6 +120,7 @@
       - id: ClothingHandsGlovesNitrile
       - id: ClothingEyesHudMedical
       - id: ClothingHeadsetAltMedical
+      - id: ClothingBeltMilitaryWebbingCMO
```
Why would CMO of all people get combat webbing?

TadJohnson00 replied:
As written in the description, its intended purpose was to allow the CMO - or whoever they choose to give it to - to carry other items in their belt slot, such as crowbars, handheld crew monitors, or shakers/beakers. The reason why I chose webbing instead of a belt is mostly for the purpose of aesthetics, I could simply resprite it if necessary. Or, if you wish for it to be removed, I can arrange that also.

Elijahrane replied:
Reduce storage space from 60 to like 30 and it's fine imo

TadJohnson00 replied:
Then consider it done.

Elijahrane commented on Resources/Prototypes/Nyanotrasen/Catalog/Fills/Items/belt.yml:
```diff
@@ -0,0 +1,9 @@
+  components:
+  - type: StorageFill
+    contents:
+      - id: Katana
```
Is this supposed to have 2 different swords in it or should these be an orGroup?

TadJohnson00 replied:
It is supposed to have both a katana and a wakizashi.

Elijahrane commented on Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:
```diff
@@ -34,4 +34,49 @@
\ No newline at end of file
+    sprite: Nyanotrasen/Clothing/Belt/Martial/black.rsi
+- type: entity
+
```
line spacing is bad here

TadJohnson00 replied:
I see, will fix.

TadJohnson00 committed:
Updated belts.yml

Changed capacity of CMO webbing from 60 to 30

Fixed spacing between Martial Artist belt and Katana belt

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Catalog/Fills/Boxes/ammunition.yml:
```diff
@@ -11,3 +11,58 @@
         - state: shellslug
+
+- type: entity
+  name: box of .25 caseless Universal magazines
```
What is the purpose of `Universal` here? It's capitalized, so I assume it means something different than `universal`.

TadJohnson00 replied:
Originally, the pistol was going to be called "Universal", after the Mk23/USP (universal self-loading pistol), but I later decided to combine the two ideas with the name "mk 32 Universal". I'll ensure that all references to the Universal are capitalised.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:
```diff
@@ -34,4 +34,49 @@
+- type: entity
+  parent: ClothingBeltStorageBase
+  id: ClothingBeltKatanaSheath
+  name: katana sheaths
```
Is there a better name than `sheaths`?

TadJohnson00 replied:
If there is, I'd be open to suggestion.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Ammunition/Magazine/caseless_rifle.yml:
```diff
@@ -0,0 +1,108 @@
+- type: entity
+  id: BaseMagazineUniversalCaselessRifle
+  name: "universal magazine (.25 caseless)"
```
`universal` here but `Universal` elsewhere?

TadJohnson00 replied:
Will fix.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:
```diff
@@ -0,0 +1,49 @@
+- type: entity
+  name: Mark 1 Rifle
```
Capitalization inconsistencies with other weapons.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:
```diff
@@ -0,0 +1,49 @@
+      - CartridgeLightRifle
+
+- type: entity
+  name: Ceremonial Rifle
```
Capitalization.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Melee/sword.yml:
```diff
@@ -0,0 +1,26 @@
+  name: wakizashi
+  parent: BaseItem
+  id: Wakizashi
+  description: Essentially a smaller katana. Fine plasteel craftwork.
```
It's funny that the 2nd paragraph of Wikipedia describes them so:
`Wakizashi are not necessarily just a smaller version of the katana; they could be forged differently and have a different cross section.[[5]](https://en.wikipedia.org/wiki/Wakizashi#cite_note-5)`
Description could be better.

TadJohnson00 replied:
I'll write something up.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Clothing/Belt/belts.yml:
```diff
@@ -34,4 +34,49 @@
+
+- type: entity
+  parent: ClothingBeltStorageBase
+  id: ClothingBeltMilitaryWebbingCMO
```
Not a fan of giving military hardware to roles that don't need it.

TadJohnson00 replied:
As before, I used webbing instead of a belt mostly due to aesthetics. I also reduced the capacity of the belt per Rane's request. If you would like me to implement further changes, I can do so.

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Pistols/pistols.yml:
```diff
@@ -0,0 +1,41 @@
+- type: entity
+  name: mk 32 "Universal"
```
Is the `Universal` necessary here? The mk 58 is just `mk 58`.

TadJohnson00 replied:
As mentioned before, the mk 32 was intended to reference the Mk23/USP, thus the name "Universal".

Vordenburg commented on Resources/Prototypes/Nyanotrasen/Entities/Objects/Weapons/Guns/Snipers/snipers.yml:
```diff
@@ -0,0 +1,49 @@
+  name: Mark 1 Rifle
+  parent: BaseWeaponSniper
+  id: WeaponSniperGrand
+  description: An ancient semi-automatic rifle that sparks some kind of a deep feeling from within you. Uses .30 rifle ammo.
```
Telling the reader what they feel is generally bad writing. Write to evoke the feeling.

TadJohnson00 replied:
I'll write up something a little nicer.

Vordenburg commented on Resources/Prototypes/Entities/Objects/Weapons/Guns/Shotguns/shotguns.yml:
```diff
@@ -128,6 +128,7 @@
   - type: Gun
     fireRate: 2
   - type: BallisticAmmoProvider
+    autoCycle: true
```
To be clear, this means the weapon no longer needs the user to click on it to eject a shell, with this set to true? They can just fire one shot after another?

TadJohnson00 replied:
That's correct, the weapon no longer needs to be cycled manually and the user can fire both barrels consecutively.

TadJohnson00 committed:
Minor fixes

- Edited a few mis-capitalisations
- Changed a couple of descriptions
- Removed erroneously added RSI
